### PR TITLE
attempting to remove some speed issues

### PR DESCRIPTION
**Pull request by williamFalcon**
trying to match exact PT speed

### CODE REVIEW

File: benchmarks/test_trainer_parity.py [hunk 1]
```diff
@@ -64,6 +64,21 @@ def test_pytorch_parity(tmpdir):
     for pl_out, pt_out in zip(lightning_outs, manual_outs):
         np.testing.assert_almost_equal(pl_out, pt_out, 5)
 
+    assert_speed_parity(pl_times, pl_times, num_epochs)
+
+
+def assert_speed_parity(pl_times, pt_times, num_epochs):
```
Comment (justusschock) [inline benchmarks/test_trainer_parity.py:70]: do we really need this two times? Can't we just import the other one?

File: pytorch_lightning/trainer/logging.py [hunk 2]
```diff
@@ -112,10 +112,6 @@ def process_output(self, output, train=False):
             num_gpus = self.num_gpus
             callback_metrics = self.reduce_distributed_output(callback_metrics, num_gpus)
 
-        for k, v in callback_metrics.items():
```
Comment (justusschock) [inline pytorch_lightning/trainer/logging.py:115]: why remove this?

File: benchmarks/test_rnn_parity.py [hunk 3]
```diff
@@ -64,6 +64,21 @@ def test_pytorch_parity(tmpdir):
     for pl_out, pt_out in zip(lightning_outs, manual_outs):
         np.testing.assert_almost_equal(pl_out, pt_out, 8)
 
+    assert_speed_parity(pl_times, pl_times, num_epochs)
```
Comment (Borda) [inline benchmarks/test_rnn_parity.py:67]:
```suggestion
    assert_speed_parity(pl_times, pt_times, num_epochs)
```

File: benchmarks/test_trainer_parity.py [hunk 4]
```diff
@@ -64,6 +64,21 @@ def test_pytorch_parity(tmpdir):
     for pl_out, pt_out in zip(lightning_outs, manual_outs):
         np.testing.assert_almost_equal(pl_out, pt_out, 5)
 
+    assert_speed_parity(pl_times, pl_times, num_epochs)
```
Comment (Borda) [inline benchmarks/test_trainer_parity.py:67]:
```suggestion
    assert_speed_parity(pl_times, pt_times, num_epochs)
```

File: pytorch_lightning/trainer/logging.py (hunk 2 above)
Comment (williamFalcon) [reply pytorch_lightning/trainer/logging.py:115]: it’s done twice. this is the first time but it’s done again at the end of the method

File: benchmarks/test_trainer_parity.py (hunk 1 above)
Comment (williamFalcon) [reply benchmarks/test_trainer_parity.py:70]: fair, i was just trying to find the threshold number before finalizing pr. will refactor this

**Pull request merged**